<launch>
    <group ns = "yolov5_ros">
        <!-- Detection configuration -->
        <arg name="weights" default="/home/sobits/catkin_ws/src/robocup_opl_tidyup/yolov5_ros/src/yolov5/weights/ak/best.pt"/>
        <arg name="data" default="$(find yolov5_ros)/src/yolov5/data/coco128.yaml"/>
        <arg name="confidence_threshold" default="0.75"/>
        <arg name="iou_threshold" default="0.45"/>
        <arg name="maximum_detections" default="1000"/>
        <arg name="device" default="0"/>
        <arg name="agnostic_nms" default="true"/>
        <arg name="line_thickness" default="3"/>
        <arg name="dnn" default="true"/>
        <arg name="half" default="false"/>

        <!-- replace imgsz -->
        <arg name="inference_size_h" default="1280"/>
        <arg name="inference_size_w" default="720"/>
        <!-- <arg name="inference_size_h" default="480"/>
        <arg name="inference_size_w" default="640"/> -->

        <!-- Visualize using OpenCV window -->
        <arg name="view_image" default="false"/>

        <!-- ROS topics -->
        <arg name="input_image_topic" default="/rgb/image_raw"/>
        <!-- <arg name="input_image_topic" default="/camera/color/image_raw"/> -->
        <!-- <arg name="input_image_topic" default="/hand_camera/color/image_raw"/> -->
        <arg name="output_topic" default="objects_rect"/>

        <!-- Optional topic (publishing annotated image) -->
        <arg name="publish_image" default="true"/>
        <arg name="output_image_topic" default="/yolov5/image_out"/>


        <node pkg="yolov5_ros" name="detect" type="detect.py" output="screen">
            <param name="weights" value="$(arg weights)"/>
            <param name="data" value="$(arg data)"/>
            <param name="confidence_threshold" value="$(arg confidence_threshold)"/>
            <param name="iou_threshold" value="$(arg iou_threshold)" />
            <param name="maximum_detections" value="$(arg maximum_detections)"/>
            <param name="device" value="$(arg device)" />
            <param name="agnostic_nms" value="$(arg agnostic_nms)" />
            <param name="line_thickness" value="$(arg line_thickness)"/>
            <param name="dnn" value="$(arg dnn)"/>
            <param name="half" value="$(arg half)"/>

            <param name="inference_size_h" value="$(arg inference_size_h)"/>
            <param name="inference_size_w" value="$(arg inference_size_w)"/>

            <param name="input_image_topic" value="$(arg input_image_topic)"/>
            <param name="output_topic" value="$(arg output_topic)"/>

            <param name="view_image" value="$(arg view_image)"/>

            <param name="publish_image" value="$(arg publish_image)"/>
            <param name="output_image_topic" value="$(arg output_image_topic)"/>
    </node>
    <!-- <include file="$(find camera_launch)/launch/d435.launch"/> -->
        <!--YOLOv3 3D-->
        <node name="object_pose_publisher_node" pkg="sobit_common_library" type="object_pose_publisher_v2" output="screen"/>
        <!-- <param name="obj_under_rate" type="double" value="0.6"/> -->
        <!-- <param name="obj_top_rate" type="double" value="0.9"/> -->
        <!-- <param name="map_frame_name" type="string" value="base_footprint"/> -->
        <param name="map_frame_name" type="string" value="azure_camera_base"/>
        <!-- <param name="base_frame_name" type="string" value="base_footprint"/> -->
        <param name="base_frame_name" type="string" value="azure_camera_base"/>
        <param name="cloud_topic_name" type="string" value="/points2"/>
        <!-- <param name="cloud_topic_name" type="string" value="/camera/depth_registered/points"/> -->
        <param name="img_topic_name" type="string" value="/rgb/image_raw"/>
        <!-- <param name="img_topic_name" type="string" value="/camera/color/image_raw"/> -->
        <param name="execute_default" type="bool" value="false"/>
        <!-- <param name="cloud_topic_name" type="string" value="/hand_camera/depth/color/points"/>
        <param name="img_topic_name" type="string" value="/hand_camera/color/image_raw"/> -->
        <!-- <param name="max_distance_to_object" type="double" value="2.0"/> -->
        <!-- <param name="image_width" type="double" value="1024"/> -->
        <!-- <param name="diff_x" type="double" value="-2.0"/> -->
        <!-- <param name="diff_y" type="double" value="0.0"/> -->
        <!-- <param name="diff_z" type="double" value="0.0"/> -->
        <!-- <param name="cluster_tolerance" type="double" value="0.03"/> -->
        <!-- <param name="min_clusterSize" type="int" value="100"/> -->
        <!-- <param name="max_clusterSize" type="int" value="5000"/> -->
        <!-- <param name="leaf_size" type="double" value="0.01"/> -->
    </group>

</launch>
